src/averageAges.js

```diff
@@ -15,11 +15,33 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let count = 0;
+  let result;
+
+  if (century) {
+    result = people.reduce((a, b) => {
+      if (b.sex === 'm' // Sex is male && death in the specified century
+            && (Math.ceil(b.died / 100) === century)) {
+        count++;
+
+        return a + b.died - b.born;
+      } else {
+        return a;
+      }
+    }, 0);
+  } else {
+    result = people.reduce((a, b) => {
+      if (b.sex === 'm') { // Sex is male
+        count++;
+
+        return a + b.died - b.born;
+      } else {
+        return a;
+      }
```

Comment:
if is not a good idea to have nested ifs because readability is not so good
here you need only filter and map methods